README.md

```diff
@@ -120,6 +120,17 @@ Then there are no required options.
 GoCD agent will use a [List S3 Objects](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html) request to determine which objects to download. If `SubPath` is specified, then only objects starting with a `SubPath` will be downloaded.
 If no files match, then fetch task fails.
 
+## Expanding environment variables
+
+The `Destination` of the publish artifact config supports expanding environment variables available within the GoCD job context.
+That means that you can specify destination S3 prefix using e.g. `${GO_PIPELINE_NAME}/my/path/${GO_PIPELINE_COUNTER}`.
+
+One of the most important uses of this feature is when destination directory is specified by `${GO_PIPELINE_NAME}/${GO_PIPELINE_COUNTER}/${GO_STAGE_NAME}/${GO_STAGE_COUNTER}/${GO_JOB_NAME}`.
+Such destination is unique for each build. It is the same tree structure which original GoCD artifact store uses.
+This plugin provides a convenience `${GO_ARTIFACT_LOCATOR}` which evaluates to above expression.
```

Comment:
Suggestion here: the behavior for GoCD is creating this destination directory without the user specifying it.

In order to keep some consistency of behavior, shall we do the same here? Although if we do then the user has less freedom on this :) Would just like to hear your thoughts @tomzo 👍

Reply:
Yeah, I thought about it. I don't feel strongly way or another.
Now the plugin doesn't really behave exactly like GoCD did. Now just in this case.
 
But this is more powerful too. It can help implement other routines which previously were hard to do and required custom scripts.

I'd start with the way it is. With most freedom for the user. Maybe later we could add an option in artifact store configuration which would pre-append this to all destinations?

Reply:
Yup, seems fair enough to me. Just wanted to understand your thoughts about this :) All seems great to me